File: LeetCode/[M] - 452. Minimum Number of Arrows to Burst Balloons/solution.go
package main

import "slices"

func findMinArrowShots(points [][]int) int {
	slices.SortFunc(points, func(s1, s2 []int) int {
		return s1[1] - s2[1]
	})
	xArr := points[0][1]
	res := 1
	for i := 1; i < len(points); i++ {
		xS, xE := points[i][0], points[i][1]
		if xS > xArr {
			xArr = xE
			res += 1
		}
	}
	return res
}

File: LeetCode/[M] - 452. Minimum Number of Arrows to Burst Balloons/solution.py
from typing import List


class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[1])
        xArr = points[0][1]
        res = 1
        for i in range(1, len(points)):
            xS, xE = points[i][0], points[i][1]
            if xS > xArr:
                xArr = xE
                res += 1
        return res
